controllers/toolchaincluster/toolchaincluster_controller.go

```diff
@@ -75,20 +78,52 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		reqLogger.Error(err, "cannot create ClientSet for the ToolchainCluster")
 		return reconcile.Result{}, err
 	}
-	healthChecker := &HealthChecker{
-		localClusterClient:     r.Client,
-		remoteClusterClient:    cachedCluster.Client,
-		remoteClusterClientset: clientSet,
-		logger:                 reqLogger,
+
+	// execute healthcheck
+	healthcheckResult, err := r.runHealthcheck(ctx, r.Client, cachedCluster.Client, clientSet, toolchainCluster, reqLogger)
+	if err != nil {
+		reqLogger.Error(err, "unable to run healthcheck for ToolchainCluster")
+		return reconcile.Result{}, err
 	}
 	// update the status of the individual cluster.
-	if err := healthChecker.updateIndividualClusterStatus(ctx, toolchainCluster); err != nil {
+	if err := r.updateStatus(ctx, toolchainCluster, healthcheckResult); err != nil {
 		reqLogger.Error(err, "unable to update cluster status of ToolchainCluster")
 		return reconcile.Result{}, err
 	}
-
 	return reconcile.Result{RequeueAfter: r.RequeAfter}, nil
 }
+func (r *Reconciler) runHealthcheck(ctx context.Context, lcl client.Client, rcl client.Client, rclset *kubeclientset.Clientset, tc *toolchainv1alpha1.ToolchainCluster, lgr logr.Logger) ([]toolchainv1alpha1.Condition, error) {
```

Comment:
the `tc *toolchainv1alpha1.ToolchainCluster` is not used anywhere.
the logger can be also dropped - we can always construct the logger from the context

Reply:
yep done